crates/holochain_serialized_bytes/Cargo.toml

```diff
@@ -12,7 +12,7 @@ repository = "https://github.com/holochain/holochain-serialization"
 edition = "2018"
 
 [dependencies]
-serde = { version = "1.0.123", features = ["serde_derive"] }
+serde = { version = "1.0.181", features = ["serde_derive"] }
```

Comment:
This isn't pinned, do we want to set this as a maximum version and only upgrade intentionally?

Reply:
Probably ought to come with an automated job to do the bumps though, so maybe not something for this PR

Reply:
I didn't pin this, because the tests will prevent unwanted serialization changes like the one that happened. But I'm open to pinning it.

Reply:
I agree the tests will catch it which is why I think there's value in automating the bump-check-merge process for accepting new serde versions

Reply:
I don't understand your suggestion. The "API test" runs on CI, so whenever there's a bump, be it intended or not, the check is done automatically. What are you proposing beyond that?

Reply:
What I have in my head is
- Pin serde here
- Set up a workflow that will run when serde does a release
- Bump the pinned dependency
- Run the tests
- If they pass, bump the holochain serialisation version
- Publish the the holochain serialisation library
- Don't pin the serialisation library in Holochain where it is consumed but assume that updates are safe to consume.

Because otherwise the CI here only runs when we make a change. So if serde releases 1.0.182 with a breaking change right now, we'll still be pinned to a holochain serialisation version in Holochain, but this permits updates. So Holochain will get 1.0.182 of serde. We'll only notice a problem when we go to bump the client libraries for Holochain or even when existing hApps are installed onto the new version of Holochain. Which is too late because we've released it by then.

This is my summary of the chain of problems that led to the previous serde issue.

I don't think this needs to be done on this PR but I think we do need to control library updates that tightly for something this critical to communication between components that are released/built/source controlled separately.